addon/interfaces/index.ts

```diff
@@ -5,4 +5,9 @@ export interface IMap<K extends object, V> {
   set(key: K, value: V): this;
 }
 
+export interface IDisposable {
```

Comment:
Sorry for bikeshedding here, but I think the name doesn't quite work. Specifically, we _already_ use this terminology ("disposable") in ember-lifeline for a very different thing. In [registerDisposable](https://github.com/ember-lifeline/ember-lifeline#registerdisposable) we are calling the callback function to run a "disposable" (not the object argument itself).

Maybe we could use `IDestroyable` here (which matches the boolean property names better anyways)?

Reply:
I thought of this, but opted for `IDisposable` as I feel like it implies that it's able to be disposed. I can get the argument that the disposable is _actually_ the function getting called, not the object. The object is something that can be destroyed.

I think I've just argued the case for myself to change it...